openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -261,3 +267,69 @@ def post(self, request, usage_key_str):
         except ValueError as err:
             raise ValidationError(detail=str(err))
         return Response(LibraryXBlockOlxSerializer({"olx": new_olx_str}).data)
+
+
+@view_auth_classes()
+class LibraryBlockAssetListView(APIView):
+    """
+    Views to list an existing XBlock's static asset files
+    """
+    @convert_exceptions
+    def get(self, request, usage_key_str):
+        """
+        List the static asset files belonging to this block.
+        """
+        key = LibraryUsageLocatorV2.from_string(usage_key_str)
+        files = api.get_library_block_static_asset_files(key)
+        return Response(LibraryXBlockStaticFilesSerializer({"files": files}).data)
+
+
+@view_auth_classes()
```

Comment:
Is this sufficient for permissions checking on these views?

Reply:
> Is this sufficient for permissions checking on these views?

Not really; this is the bare minimum. I've got a couple pending PRs that will flesh that out fully.

Reply:
WIP PR for permissions is here: https://github.com/edx/edx-platform/pull/22258
(Currently includes some commits from this branch though)